pkg/cmd/download.go

```diff
@@ -30,42 +31,45 @@ import (
 )
 
 // NewDownloadCmd returns a new download command.
-func NewDownloadCmd() *cobra.Command {
+func NewDownloadCmd(targetReader TargetReader) *cobra.Command {
 	return &cobra.Command{
 		Use:   "download tf + (infra|internal-dns|external-dns|ingress|backup)\n  gardenctl download logs vpn\n ",
 		Short: "Download terraform configuration/state for local execution for the targeted shoot or log files",
-		Run: func(cmd *cobra.Command, args []string) {
+		RunE: func(cmd *cobra.Command, args []string) error {
 			if len(args) != 2 || !(args[1] == "infra" || args[1] == "internal-dns" || args[1] == "external-dns" || args[1] == "ingress" || args[1] == "backup" || args[1] == "vpn") {
-				fmt.Println("Command must be in the format:\n  download tf + (infra|internal-dns|external-dns|ingress|backup)\n  download logs vpn")
-				os.Exit(2)
+				return errors.New("Command must be in the format:\n  download tf + (infra|internal-dns|external-dns|ingress|backup)\n  download logs vpn")
 			}
 			switch args[0] {
 			case "tf":
-				path := downloadTerraformFiles(args[1])
+				path := downloadTerraformFiles(args[1], targetReader)
 				fmt.Println("Downloaded to " + path)
 			case "logs":
-				downloadLogs(args[1])
+				downloadLogs(args[1], targetReader)
 			default:
 				fmt.Println("Command must be in the format:\n  download tf + (infra|internal-dns|external-dns|ingress|backup)\n  download logs vpn")
 			}
+			return nil
 		},
 		ValidArgs: []string{"tf"},
 	}
 }
 
 // downloadTerraformFiles downloads the corresponding tf file
-func downloadTerraformFiles(option string) string {
+func downloadTerraformFiles(option string, targetReader TargetReader) string {
 	namespace := ""
-	var target Target
-	ReadTarget(pathTarget, &target)
+	target := targetReader.ReadTarget(pathTarget)
 	// return path allow non operator download key file
 	if getRole() == "user" {
-		gardenName := target.Stack()[0].Name
-		projectName := target.Stack()[1].Name
 		if (len(target.Stack()) < 3) || (len(target.Stack()) == 3 && target.Stack()[2].Kind == "namespace") {
 			fmt.Println("No Shoot targeted")
 			os.Exit(2)
+		} else if target.Stack()[1].Kind == "seed" {
```

Comment:
first of all, we don't have any guidelines in README to encourage `user` role to target -shoot via -seed,
secondly, the `user` role can't see the seed information form cluster WebUI as default.  In this way, I think the `user` role may don't have any chance to target seed in any case. so I am assuming this code maybe never executing for `user` role

Reply:
1) I'm not saying we have anything `encourage` user action to target shoot, in line 64 `projectName := target.Stack()[1].Name` i don't think it's proper as we don't check whether target.Stack()[1] is seed or project but just directly assume it's project
2) in gardenctl , the `target` is read from a file , anyone including user role could modify the file to make the garden target include garden/seed/shoot, the user role is only determined during run time api call, so this piece code of check is necessary

Reply:
Besides, if you really want to add msg I think it should be in `target.go` file for `targetseed` part. as it is the beginning of the process when `user` role operator any task.

but as mentioned at the above, not seed information,  I guess, this code maybe not executing for the `user` role

Reply:
> Besides, if you really want to add msg I think it should be in `target.go` file for `targetseed` part. as it is the beginning of the process when `user` role operator any task.

this is a good point for enhancement of `target.go` but out of this PR scope

> but as mentioned at the above, not seed information, I guess, this code maybe not executing for the `user` role

as i mentioned above, anyone have access to local file got a chance to set target like `garden/seed/shoot`

Reply:
To generate the local file is base on rules from `target.go` and the input from user. It should be handle in there at the begging for everything not just fix in the `download.go`. That why I say the code in there is not necessary.  why not we just block and display the warming information in `user` role when they target seed at the begging instead of fix in `download.go`

Reply:
refactor `target.go` will not prevent from anyone modifying local file directly

Reply:
why people modifying local file directly ? besides, the logic for all the gardenctl reload from `.garden/sessions/plantingSession/target` ,  everytime when `user` role `target -g garden -p project -t shoot` will overwrite it by target.go.

Reply:
why can't ???

Reply:
> why can't ???

My suggestion is if you need to check `seed` you can add in `target.go` at the beginning of the process.

Reply:
> > why can't ???
> 
> My suggestion is if you need to check `seed` for `user` role you can add in `target.go` at the beginning of the process.

i don't think we are on same page here.
To Summary:
- for enhancement of target.go i've already opened new issue https://github.com/gardener/gardenctl/issues/309
- what i'm talking about is that you can't prevent people from editing target file ->you never know whether it's `g/s/s` or `g/p/s` in target stack -> you need to check this, that's what my PR stands for.